meta-mentor-staging/classes/sdk_auto_relocate.bbclass

```diff
@@ -17,23 +17,31 @@ SDK_POST_INSTALL_COMMAND_append = 'echo "${env_setup_script%/*}" >"${env_setup_s
 sdkpath_to_bindir = "${@os.path.relpath('${SDKPATHNATIVE}${bindir_nativesdk}', '${SDKPATH}')}"
 
 toolchain_env_script_reloc_fragment () {
-    cat <<END >>"$script"
+    mv "$script" "$script.fragment"
+    cat >"$script" <<END
 if [ -z "\$SDK_RELOCATING" ]; then
     if [ -n "\$BASH_SOURCE" ] || [ -n "\$ZSH_NAME" ]; then
         if [ -n "\$BASH_SOURCE" ]; then
             scriptdir="\$(cd "\$(dirname "\$BASH_SOURCE")" && pwd)"
         elif [ -n "\$ZSH_NAME" ]; then
             scriptdir="\$(cd "\$(dirname "\$0")" && pwd)"
         fi
-        env -i "\$scriptdir/${sdkpath_to_bindir}/sdk-auto-relocate" && SDK_RELOCATING=1 . "\$scriptdir/environment-setup-${REAL_MULTIMACH_TARGET_SYS}"
-        unset scriptdir
+
+        if [ -e "\$scriptdir/sysroots/i686-melsdk-linux/usr/bin/sdk-auto-relocate" ]; then
```

Comment:
This hardcodes the sysroot path when the path one line below it does not. Please copy/paste the correct version that doesn't hardcode i686-melsdk-linux :)